compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -1593,6 +1593,12 @@ void OMR::Power::MemoryReference::accessStaticItem(TR::Node *node, TR::SymbolRef
          loadAddressConstant(cg, true, nodeForSymbol, 1, reg, NULL, false, TR_BodyInfoAddressLoad);
          return;
          }
+      else if (symbol->isCatchBlockCounter() && cg->comp()->compileRelocatableCode())
```

Comment:
It's better to replace `compileRelocatableCode()` with something like `needRelocationsForBodyInfoData()`